### PR TITLE
Added tests for a Dialog

### PR DESCRIPTION
**Pull request by krolwofficial**
Added tests for a Dialog

### CODE REVIEW

File: src/alto-ui/Dialog/__tests__/Dialog.test.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  return props => {
+    return <button className="CloseButton" onClick={() => props.onClick()} />;
+  };
+});
+
+describe('Dialog', () => {
+
+  const defaultProps = {
+    id: 'string',
+    // className: 'string',
```
Comment (barteksiudak) [inline src/alto-ui/Dialog/__tests__/Dialog.test.js]: If you don't need those than remove all comments

File: src/alto-ui/Dialog/__tests__/Dialog.test.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Dialog from '../Dialog';
+
+import CloseButton from '../../CloseButton';
+
+jest.mock('../../CloseButton', () => {
+  return props => {
```
Comment (barteksiudak) [inline src/alto-ui/Dialog/__tests__/Dialog.test.js]: desctucturize props here and you don't need body of function here:
```
jest.mock('../../CloseButton', () => ({ onClick }) => (
  <button className="CloseButton" onClick={() => onClick()} />
);
```

File: src/alto-ui/Dialog/__tests__/Dialog.test.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    // open: true,
+    // title: 'string',
+    // onClose: jest.fn(),
+    // buttons: [{}, {}],
+    // closeFocusTargetId: 'string',
+    // inert: true,
+    // a11yLabelClose: 'string',
+  }
+
+  const getWrapper = ({ ...props }) => shallow(<Dialog {...defaultProps} {...props} />);
```
Comment (barteksiudak) [inline src/alto-ui/Dialog/__tests__/Dialog.test.js]: why `({ ...props })` ?
just:
```
const getWrapper = props => shallow(<Dialog {...defaultProps} {...props} />);
```

File: src/alto-ui/Dialog/__tests__/Dialog.test.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+  it('is Dialog',() => {
+    const wrapper = getWrapper({
+      id: 'key'
+    });
+    expect(wrapper.find('.Dialog__overlay').exists()).toBe(true);
+  })
+
+  it('should render div with Dialog className',() => {
```
Comment (barteksiudak) [inline src/alto-ui/Dialog/__tests__/Dialog.test.js]: The comment is not clear :(
it("should has element with Dialog className")

File: src/alto-ui/Dialog/__tests__/Dialog.test.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    const props = {
+      open: true,
+    }
+
+    const wrapper = getWrapper(props);
+    expect(wrapper.find('.Dialog').exists()).toBe(true);
+  })
+
+  it('should render header with Dialog__header className',() => {
```
Comment (barteksiudak) [inline src/alto-ui/Dialog/__tests__/Dialog.test.js]: should has header element

File: src/alto-ui/Dialog/__tests__/Dialog.test.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    const propsClose = {
+      open: true,
+      onClose: jest.fn()
+    }
+
+    expect(getWrapper(propsTitle).find('.Dialog__header').exists()).toBe(true);
+    expect(getWrapper(propsClose).find('.Dialog__header').exists()).toBe(true);
+  })
+
+  it('should render h2 with Dialog__title className',() => {
```
Comment (barteksiudak) [inline src/alto-ui/Dialog/__tests__/Dialog.test.js]: should has and write negative test too

**Comment by barteksiudak** [comment]
There missing tests for buttons!
you should test how many buttons exits and if those are clickable.
Check if `flat` prop for randomly chosen buttons are passed correctly and this same for `disabled` prop

File: src/alto-ui/Dialog/__tests__/Dialog.test.js (hunk 6 above)
Comment (krolwofficial) [reply src/alto-ui/Dialog/__tests__/Dialog.test.js]: I wrote negative tests:
`open: false`  =>  `it('should has Overlay element')`
`title: '', onClose: undefined`  => `it('should no has element with Dialog__header className')`

File: src/alto-ui/Dialog/__tests__/Dialog.test.js (hunk 5 above)
Comment (krolwofficial) [reply src/alto-ui/Dialog/__tests__/Dialog.test.js]: Refactored to `it(`should has element with Dialog__header className`)`

File: src/alto-ui/Dialog/__tests__/Dialog.test.js (hunk 4 above)
Comment (krolwofficial) [reply src/alto-ui/Dialog/__tests__/Dialog.test.js]: Refactored to `it('should has element with Dialog className')`

File: src/alto-ui/Dialog/__tests__/Dialog.test.js (hunk 3 above)
Comment (krolwofficial) [reply src/alto-ui/Dialog/__tests__/Dialog.test.js]: Refactored

File: src/alto-ui/Dialog/__tests__/Dialog.test.js (hunk 2 above)
Comment (krolwofficial) [reply src/alto-ui/Dialog/__tests__/Dialog.test.js]: Destructurized

File: src/alto-ui/Dialog/__tests__/Dialog.test.js (hunk 1 above)
Comment (krolwofficial) [reply src/alto-ui/Dialog/__tests__/Dialog.test.js]: Removed

File: src/alto-ui/Dialog/__tests__/Dialog.test.js [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+jest.mock('../../CloseButton', () => ({onClick}) => (
+  <button className="CloseButton" onClick={() => onClick()} />
+));
+
+jest.mock('../../Button', () => ({onClick}) => (
+  <button
+    className="Dialog__button"
+    onClick={() => onClick()}
+    // flat={index < buttons.length - 1}
```
Comment (djdeimon) [inline src/alto-ui/Dialog/__tests__/Dialog.test.js]: remove commented code

Comment (krolwofficial) [reply src/alto-ui/Dialog/__tests__/Dialog.test.js]: Removed, I refactored the mock declarations

**Pull request merged**